test/integration/smart_proxy_test.rb

```diff
@@ -39,4 +39,47 @@ class SmartProxyIntegrationTest < ActionDispatch::IntegrationTest
     assert page.has_selector?('h3', :text => "DHCP")
     assert page.has_content? "Version"
   end
+
+  test "show page with conditional pagelets" do
+    Pagelets::Manager.add_pagelet("smart_proxies/show", :main_tabs, :name => "Visible", :partial => "../../test/static_fixtures/views/test", :onlyif => Proc.new { true })
+    Pagelets::Manager.add_pagelet("smart_proxies/show", :main_tabs, :name => "Hidden", :partial => "../../test/static_fixtures/views/test", :onlyif => Proc.new { false })
+    proxy = smart_proxies(:one)
+    visit smart_proxy_path(proxy)
+    assert page.has_link? "Visible"
+    refute page.has_link? "Hidden"
+  end
+
+  test "show page passes @smart_proxy into pagelets" do
+    Pagelets::Manager.add_pagelet("smart_proxies/show", :main_tabs,
+                                                        :name => "VisibleTab",
+                                                        :partial => "../../test/static_fixtures/views/test",
+                                                        :onlyif => Proc.new { |model| model.has_feature? "DHCP" })
+    Pagelets::Manager.add_pagelet("smart_proxies/show", :main_tabs,
+                                                        :name => "HiddenTab",
+                                                        :partial => "../../test/static_fixtures/views/test",
+                                                        :onlyif => Proc.new { |model| model.has_feature? "TFTP" })
+    proxy = smart_proxies(:one)
+    visit smart_proxy_path(proxy)
+    assert page.has_link? "VisibleTab"
+    refute page.has_link? "HiddenTab"
+  end
+
+  test "show page renders basic pagelets" do
+    Pagelets::Manager.add_pagelet("smart_proxies/show", :main_tabs,
+                                                        :name => "TestTab",
+                                                        :partial => "../../test/static_fixtures/views/test_partial")
```

Comment:
@tstrachota: they have a different content, otherwise no difference. I added the second one so I could test for content of the page and not to worry about something being already in the pagelets manager.